Sources/CoreAardvarkTests/ARKLogDistributorSwiftTests.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Copyright 2021 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreAardvark
+import XCTest
+
+final class ARKLogDistributorSwiftTests: XCTestCase {
+
+    func test_log_defaultValues() throws {
+        let logDistributor = ARKLogDistributor()
+        let observer = TestObserver()
+        logDistributor.add(observer)
+
+        logDistributor.log("Hello world")
+
+        let distributeExpectation = expectation(description: "distributes log message")
+        logDistributor.distributeAllPendingLogs {
+            distributeExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5, handler: nil)
+
+        XCTAssertEqual(observer.observedLogMessages.count, 1)
+
+        let logMessage = try XCTUnwrap(observer.observedLogMessages.first)
+        XCTAssertEqual(logMessage.text, "Hello world")
+        XCTAssertEqual(logMessage.type, .default)
+        XCTAssertEqual(logMessage.image, nil)
+        XCTAssertEqual(logMessage.parameters, [:])
+        XCTAssert(logMessage.userInfo.isEmpty)
+    }
+
+    func test_log_explicitValues() throws {
+        let logDistributor = ARKLogDistributor()
+        let observer = TestObserver()
+        logDistributor.add(observer)
+
+        logDistributor.log(
+            "Hello world",
+            type: .error,
+            image: UIImage(),
+            parameters: ["Foo": "Bar"],
+            userInfo: ["Test": "Testing"]
+        )
+
+        let distributeExpectation = expectation(description: "distributes log message")
+        logDistributor.distributeAllPendingLogs {
+            distributeExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5, handler: nil)
+
+        XCTAssertEqual(observer.observedLogMessages.count, 1)
+
+        let logMessage = try XCTUnwrap(observer.observedLogMessages.first)
+        XCTAssertEqual(logMessage.text, "Hello world")
+        XCTAssertEqual(logMessage.type, .error)
+        XCTAssertEqual(logMessage.image, UIImage())
+        XCTAssertEqual(logMessage.parameters, ["Foo": "Bar"])
+        XCTAssertFalse(logMessage.userInfo.isEmpty)
+    }
+
+}
+
+// MARK: -
+
+extension ARKLogDistributorSwiftTests {
```

Comment:
not blocking, but it feels a little silly to use an extension + `fileprivate` when you could define the below object within the initial declaration above and use `private`. You can still keep the `// MARK: -`

Reply:
Fair enough